src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -37,6 +38,8 @@ public static class Options
             public const string UnreferencedTypesHandling = "unreferenced-types-handling";
             public const string ModelFactoryForHlc = "model-factory-for-hlc";
             public const string GenerateModelFactory = "generate-model-factory";
+            public const string TreatEmptyStringAsNullForModels = "treat-empty-string-as-null-for-models";
+            public const string TreatEmptyStringAsNullForExtraTypes = "treat-empty-string-as-null-for-extra-types";
```

Comment:
from its usage, it's about intrinsic types (e.g. framework types, azure core types):
```suggestion
            public const string TreatEmptyStringAsNullForIntrinsicTypes = "treat-empty-string-as-null-for-intrinsic-types";
```

Reply:
Extra means that we also have a list of default types that will treat empty string as null. Intrinsic type seems not very clear to me (just like framework type), is it a cadl concept? I found this [code snippet](https://github.com/Azure/autorest.csharp/blob/7ec42556b6515821f1f1f02b654c98b583a1cf61/src/CADL.Extension/Emitter.Csharp/src/lib/model.ts#L244-L267), it looks like number and boolean are not included in intrinsic types? But we may need to support them as long as we may convert the original string type in swagger to those types.

Reply:
How about `ForCoreTypes`?

Reply:
The name of this config sounds like it is a toogle of a feature, whether we want to treat empty string as null for those types, instead of meaning this is a list that you want to do that.
How about we change the order of words like `intrinsic-types-to-treat-empty-string-as-null`

Reply:
`TreatEmptyStringAsNullForExtraTypes` is to select the types from an intrinstic type lists to treat empty string as null. Right?
If so, I don't see that we need to separate handle this as regular type.
I think it is better to eliminate the number of options as possible.

Reply:
It looks like you guys are all familiar with the term `intrinsic types`, then I'm fine to use this name. I will use `additional-intrinsic-types-to-treat-empty-string-as-null` as the option name.

Reply:
> `TreatEmptyStringAsNullForExtraTypes` is to select the types from an intrinstic type lists to treat empty string as null. Right? If so, I don't see that we need to separate handle this as regular type. I think it is better to eliminate the number of options as possible.

The models config is to specify which models, the types config is to specify which properties of the specified models based on type match. 

This is a balance of not making too many properties (let's say based on the type match within an RP) to treat empty string as null (there is a risk that services may actually treat the two differently) and not make the config too granular (let's say specify each properties with a single config) since that may lead to fixing deserialzation issues one property by one property multiple times for the same model relying on customer report.